spec/core/instances_spec.rb

```diff
@@ -545,4 +553,100 @@
       end
     end
   end
+
+  describe '#has_string_attribute?' do
+    it 'returns false if no string attribute exists' do
+      expect(subject.has_string_attribute?).to be false
+    end
+
+    context 'dataset has string attribute' do
+      subject { load_instances('weather.string.arff') }
+
+      it 'returns true if string attribute exists' do
```

Comment:
The if part of the description can be pulled, it's already given by the surrounding context.